### ISSUE TITLE
Error: Spotify does not appear to be installed. Note: You must whitelist the 'spotify' URL scheme in your info.plist

### ISSUE THREAD

**Issue by P-Russell** [cjam/react-native-spotify-remote#104]
Hi there.

I would like my persist spotify sessions over closing and opening the app that we are building.

To achieve this I am experimenting with saving the session returned by SpotifyAuth.authorize to asyncStorage. 

When I try to reinitialise the session by passing that saved token to SpotifyRemote.connect I get this Error: Spotify does not appear to be installed. Note: You must whitelist the 'spotify' URL scheme in your info.plist

My info.plist does include the whitelisting as per the IOS SDK docs:
```
<key>LSApplicationQueriesSchemes</key>
<array>
    <string>spotify</string>
</array>
```

**Comment by cjam** [comment]
Hmm interesting.  I haven't actually gotten around to this use case yet but I've been hoping to.  A few questions.

Is this happening on iOS / Android or both?

Which version of the package are you using?

**Comment by denis0324** [comment]
Same issue.
I am using react-native-spotify-remote v0.3.5
I am getting accessToken throught `auth.getSession()` and `auth.authorize()`, but I am facing this issue when I call `remote.connect()`

FYI are there any way to get user playlists (`v1/me/playlists`) in this library?

**Comment by codeCasa** [comment]
> Same issue on iOS.
> I am using react-native-spotify-remote v0.3.5
> I am getting accessToken throught `auth.getSession()` and `auth.authorize()`, but I am facing this issue when I call `remote.connect()`
> 
> FYI are there any way to get user playlists (`v1/me/playlists`) in this library?

What I've done as far as getting user playlists and other data is take the session access token and couple it with the web api, its bit more work, but it does offer a wider set of tools.

**Comment by codeCasa** [comment]
@P-Russell 

```

scopes: [
            ApiScope.AppRemoteControlScope,
            ApiScope.StreamingScope,
            ApiScope.PlaylistModifyPrivateScope,
            ApiScope.PlaylistReadPrivateScope,
            ApiScope.PlaylistModifyPublicScope,
        ]

...
    public connect = async (): Promise<boolean> => {
        try {
            await SpotifyAuth.endSession();
            const session = await SpotifyAuth.authorize(this.spotifyConfig);
            this.appSettings.saveBoolean(Settings.DidConnectToSpotify, true);
            await this.appSettings.saveSecureAsync(Settings.SpotifyAccessToken, session.accessToken);
            await this.appSettings.saveSecureAsync(Settings.SpotifyRefreshToken, session.refreshToken);
           ...
            return true;
        } catch (err) {
            return false;
        }
    };

    public connectRemote = async (): Promise<boolean> => {
        try {
            if (await SpotifyRemote.isConnectedAsync()) {
                return true;
            }
            const tokenString = await this.appSettings.getSecureAsync(Settings.SpotifyAccessToken);
            if (!tokenString || tokenString.length === 0) {
                return false;
            }
            await SpotifyRemote.connect(tokenString);
            return await SpotifyRemote.isConnectedAsync();
        } catch (ex) {
            console.log("failed to connect", ex);
            return false;
        }
    };

```

Also it seems that Spotify must be actively playing music if your app is to connect while it is in the background.

**Comment by denis0324** [comment]
One question more.
Should I add `spotify` into info.plist as url scheme?

```
<key>LSApplicationQueriesSchemes</key>
<array>
    <string>spotify</string>
</array>
```

**Comment by codeCasa** [comment]
> One question more.
> Should I add `spotify` into info.plist as url scheme?
> 
> ```
> <key>LSApplicationQueriesSchemes</key>
> <array>
>     <string>spotify</string>
> </array>
> ```

Yes you should. That is what I have as well.

**Comment by denis0324** [comment]
I tried it, but still same issue.

**Comment by cjam** [comment]
hey @denis0324, the new Spotify SDK's (i.e. the ones that this library wraps) provide an api to control the Spotify app on the phone from another app, so you will definitely need the app installed.

> @codeCasa 
> Also it seems that Spotify must be actively playing music if your app is to connect while it is in the background.

This is a limitation in iOS in due to how it aggressively backgrounds apps that aren't doing anything (which in turn disconnects this library)

Also, in my experience as with others here.  Using this library in conjunction with the web api yields the best results.  Using this library to control the music playing and state of spotify on the phone, but using the web api to get song / playlist data.

**Comment by denis0324** [comment]
Hi, @cjam 
Thanks for your reply.

Yes, I noticed that the new Spotify SDK is very difference with old one (I mean rn-spotify-sdk)
And it seems that the remote connection is only available when the Spotify plays music. (iOS)

It's because I am getting the below issue when stop music in the Spotify.
```
Connection attempt failed. Ensure the Spotify app is installed, and try to reconnect. Stream error.
Reconnect the transport to the Spotify app. The operation couldn't be completed. Connection refused.
```
And it's succeed when play music in the Spotify.

Thanks.

**Comment by cjam** [comment]
Yea @denis0324 unfortunately it's a limitation of the underlying SDK on iOS. I generally use the authorize and play flow to wake Spotify back up if it gets backgrounded. Not sure of a better solution at this point unfortunately.

**Comment by TSB1999** [comment]
@Bhavin-hyperspace Hey, i'm also just fetching the token and trying to play the track. Did you find any solutions?

**Comment by Bhavin-hyperspace** [comment]
> 'm also just fetching the token and trying to play the track. Did you find any solutions?

@TSB1999 i am able to find any solution. Did you get same error?

**Comment by TSB1999** [comment]
@Bhavin-hyperspace actually i did manage to solve the issue. But I'm using an M1 Mac and the configuration may be slightly different. 

But basically I don't think you can use the accessToken from the web API. 

With iOS for example, you need to follow all the steps in "configure info plist". Especially step two where they talk about URL schemes. 

Essentially, you need to create a URL scheme that matches the redirectURL before the :// 

Basically, read this document (https://developer.spotify.com/documentation/ios/quick-start/) and ensure you follow the steps in "configure info plist" very carefully. 

The web api access token won't work with this library because it needs to actually be redirected back via deep links. 
So this has to be confirmed in the info plist as explained above.
